atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/TimestampStateStore.java

```diff
@@ -150,10 +150,12 @@ Optional<CommitInfo> getCommitInfo(long startTimestamp) {
     TimestampStateStoreState getStateForDiagnostics() {
         // This method doesn't need to read a thread-safe snapshot of timestampMap and livingVersions
         SortedSetMultimap<Sequence, StartTimestamp> living = TreeMultimap.create();
-        livingVersions.forEach((sequence, startTimestamps) -> living.putAll(sequence, new TreeSet<>(startTimestamps)));
+        livingVersions.forEach(
+                (sequence, startTimestamps) -> living.putAll(sequence, ImmutableSortedSet.copyOf(startTimestamps)));
+
         return ImmutableTimestampStateStoreState.builder()
-                .timestampMap(new TreeMap<>(timestampMap))
-                .livingVersions(living)
+                .timestampMap(ImmutableSortedMap.copyOf(timestampMap))
+                .livingVersions(Multimaps.unmodifiableSortedSetMultimap(living))
```

Comment:
I could not find a quick immutable impl for SortedSetMultimap so doing this. Might be too much tbh